AfterClass-io/afterclass.io-v2
# feat: enable feature flag on announcement banner

closes #135 

## Changes


- added & enabled `announcement-carousel` feature flag on posthog
- add [`posthog.identify`](https://posthog.com/docs/product-analytics/identify) on `user.id` at signin => required for feature flag conditional release
- add posthog-node package and wrapper code
- call feature flag on announcement banner component under home page

## Implementation Details



see https://posthog.com/tutorials/nextjs-analytics#setting-up-and-using-feature-flags
- posthog feature flags supports both client and server sided runtimes

### Feature Flag on posthog

ENABLED - released ONLY for 100% of user matching the email `test@smu.edu.sg`

![image](https://github.com/user-attachments/assets/7dd1a040-408f-4363-8ee0-4b603b68b4e0)


## How to Test



1. head to [preview link](https://afterclass-io-v2-git-135-feature-flags-afterclass.vercel.app/)
2. see that announcement banner doesnt render
3. login with `test@smu.edu.sg` account
4. see that announcement banner renders

## Preview / Screenshots




### Not logged in - i.e user email is not `test@smu.edu.sg`

![image](https://github.com/user-attachments/assets/c4c5e73f-cdc7-4630-a912-ab45bfe5173e)

### Logged in as `test@smu.edu.sg`

![image](https://github.com/user-attachments/assets/42b63a57-57f4-4ab1-888e-dc984ad540a9)

### Logged in as `test2@smu.edu.sg`

![image](https://github.com/user-attachments/assets/ed68d6db-8287-408f-86af-f060604de4f5)



## Checklist
